src/rdf-repository/grafter/rdf/sparql.clj

```diff
@@ -3,27 +3,45 @@
             [grafter.rdf.repository :refer [repo sparql-repo ->connection]]
             [grafter.rdf.repository :as repo]
             [grafter.rdf.io :refer [->sesame-rdf-type]]
-            [clojure.java.io :refer [resource]]))
+            [clojure.java.io :refer [resource]]
+            [clojure.string :as str]))
+
+(defn pre-process-limit-clauses
+  "look through a query string and replaces limit clauses with values supplied
+  as maps against matching SPARQL ?variable names or a limit integer"
+  [sparl-query limits]
+  (reduce (fn [memo [key val]]
+            (if-let [pattern (cond
+                               (integer? key)
+                               (str "(?i)LIMIT\\s+" key)
+
+                               (or (string? key) (keyword? key))
```

Comment:
The cond returns nil on purpose, so no string replace is attempted inside the if-let